proposals/0051-options-local-pragma.rst

```diff
@@ -0,0 +1,139 @@
+Options Local Pragma
+====================
+
+.. proposal-number:: 
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_.
+.. sectnum::
+.. contents::
+
+We propose to add functionality for switching GHC-options locally. For instance, in this code
+::
+ {-# OPTIONS_GHC -Wno-name-shadowing #-}
+ 
+ f = {-# OPTIONS_LOCAL -Wname-shadowing #-}
+   let a = () in
```

Comment:
What about changing this to the example I offered?

```
f a =
  {-# OPTIONS_LOCAL -Wname-shadowing #-}
  \a -> a + a

g a = \a -> a + a
```

Here, `f` and `g` are exactly the same except for the annotation, and `f` takes one line less.